cli/azd/cmd/up.go

```diff
@@ -52,130 +43,67 @@ func newUpCmd() *cobra.Command {
+		provisioningManager:          provisioningManager,
 	}
 }
 
 func (u *upAction) Run(ctx context.Context) (*actions.ActionResult, error) {
-	if u.flags.provisionFlags.noProgress {
-		fmt.Fprintln(
-			u.console.Handles().Stderr,
-			//nolint:lll
-			output.WithWarningFormat(
-				"WARNING: The '--no-progress' flag is deprecated and will be removed in a future release.",
-			),
-		)
-		// this flag actually isn't used by the provision command, we set it to false to hide the extra warning
-		u.flags.provisionFlags.noProgress = false
-	}
-
-	if u.flags.deployFlags.serviceName != "" {
-		fmt.Fprintln(
-			u.console.Handles().Stderr,
-			//nolint:lll
-			output.WithWarningFormat("WARNING: The '--service' flag is deprecated and will be removed in a future release."))
-	}
-
-	infra, err := u.importManager.ProjectInfrastructure(ctx, u.projectConfig)
-	if err != nil {
-		return nil, err
-	}
-	defer func() { _ = infra.Cleanup() }()
-
-	err = u.provisioningManager.Initialize(ctx, u.projectConfig.Path, infra.Options)
-	if err != nil {
-		return nil, err
-	}
-
-	startTime := time.Now()
-
-	packageAction, err := u.packageActionInitializer()
-	if err != nil {
-		return nil, err
-	}
-	packageOptions := &middleware.Options{CommandPath: "package"}
-	_, err = u.runner.RunChildAction(ctx, packageOptions, packageAction)
-	if err != nil {
-		return nil, err
+	if u.projectConfig.Workflows == nil {
+		u.projectConfig.Workflows = map[string][]*workflow.Step{}
 	}
 
-	provision, err := u.provisionActionInitializer()
-	if err != nil {
-		return nil, err
-	}
+	upWorkflow, has := u.projectConfig.Workflows["up"]
+	if !has || len(upWorkflow) == 0 {
+		err := u.provisioningManager.Initialize(ctx, u.projectConfig.Path, u.projectConfig.Infra)
+		if err != nil {
+			return nil, err
+		}
 
-	provision.flags = &u.flags.provisionFlags
-	provisionOptions := &middleware.Options{CommandPath: "provision"}
-	provisionResult, err := u.runner.RunChildAction(ctx, provisionOptions, provision)
-	if err != nil {
-		return nil, err
-	}
-
-	// Print an additional newline to separate provision from deploy
-	u.console.Message(ctx, "")
-
-	deploy, err := u.deployActionInitializer()
-	if err != nil {
-		return nil, err
+		u.projectConfig.Workflows["up"] = defaultUpWorkflow
+	} else {
+		u.console.Message(ctx, output.WithWarningFormat("WARNING: Running custom 'up' workflow from azure.yaml"))
 	}
 
-	deploy.flags = &u.flags.deployFlags
-	// move flag to args to avoid extra deprecation flag warning
-	if deploy.flags.serviceName != "" {
-		deploy.args = []string{deploy.flags.serviceName}
-		deploy.flags.serviceName = ""
-	}
-	deployOptions := &middleware.Options{CommandPath: "deploy"}
-	_, err = u.runner.RunChildAction(ctx, deployOptions, deploy)
+	workflowRunAction, err := u.workflowRunActionInitializer()
 	if err != nil {
 		return nil, err
 	}
 
-	return &actions.ActionResult{
-		Message: &actions.ResultMessage{
-			Header: fmt.Sprintf("Your application was provisioned and deployed to Azure in %s.",
```

Comment:
Sort of a bummer that we lose this output - I do think it provided value to users.  Wonder if we could keep this in some form...

Reply:
It's not really being removed - just moved to be part of the workflow command. `Workflow completed in x `